CONTRIBUTING.md

```diff
@@ -77,6 +78,60 @@ to do this is to invoke `rustc` like this:
 $ RUST_BACKTRACE=1 rustc ...
 ```
 
+## The Build System
+
+The build system for Rust is complex. It covers bootstrapping the compiler,
+running tests, building documentation and more. Unless you are familiar with
+Makefiles, I wouldn't suggest trying to understand everything going on in
```

Comment:
IMHO "can easily get lost in it" is more a function of size than advancedness, but otherwise :+1: